src/Our.Umbraco.HeadRest/HeadRest.cs

```diff
@@ -18,48 +25,57 @@ public class HeadRest
 
         internal static ConcurrentDictionary<string, HeadRestConfig> Configs = new ConcurrentDictionary<string, HeadRestConfig>();
 
-        private readonly UmbracoMapper _mapper;
-
-        public HeadRest(UmbracoMapper mapper) 
-            => _mapper = mapper;
-
-        public void ConfigureEndpoint(HeadRestOptions options)
+        public void ConfigureEndpoint(IUmbracoBuilder builder, HeadRestOptions options)
         {
-            ConfigureEndpoint("/", "/root/*[@isDoc][1]", options);
+            ConfigureEndpoint("/", "/root/*[@isDoc][1]", builder, options);
         }
 
-        public void ConfigureEndpoint(string basePath, HeadRestOptions options)
+        public void ConfigureEndpoint(string basePath, IUmbracoBuilder builder, HeadRestOptions options)
         {
-            ConfigureEndpoint(basePath, "/root/*[@isDoc][1]", options);
+            ConfigureEndpoint(basePath, "/root/*[@isDoc][1]", builder, options);
         }
 
-        public void ConfigureEndpoint(string basePath, string rootNodeXPath, HeadRestOptions options)
+        public void ConfigureEndpoint(string basePath, string rootNodeXPath, IUmbracoBuilder builder, HeadRestOptions options)
         {
-            var config = _mapper.Map<HeadRestConfig>(options);
-            config.BasePath = basePath;
-            config.RootNodeXPath = rootNodeXPath;
-            ConfigureEndpoint(config);
+            var config = new HeadRestConfig(options)
```

Comment:
It was quite tricky getting hold of the `_mapper` at this point, so I changed it to pass in the old `options` and copy the values in the constructor instead.